src/Components/Endpoints/src/DependencyInjection/WebAssemblyRazorComponentsBuilderExtensions.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Endpoints;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Extension methods to configure an <see cref="IServiceCollection"/> for WebAssembly components.
+/// </summary>
+public static class WebAssemblyRazorComponentsBuilderExtensions
+{
+    /// <summary>
+    /// Adds services to support rendering interactive WebAssembly components in a razor components
+    /// application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IRazorComponentsBuilder"/>.</param>
+    /// <param name="configure">A callback to configure <see cref="WebAssemblyComponentsEndpointOptions"/>.</param>
+    /// <returns>An <see cref="IRazorComponentsBuilder"/> that can be used to further customize the configuration.</returns>
+    public static IRazorComponentsBuilder AddWebAssemblyComponents(this IRazorComponentsBuilder builder, Action<WebAssemblyComponentsEndpointOptions>? configure = null)
```

Comment:
Why does this not go on `Webassembly.Server`?

Reply:
😄 https://github.com/dotnet/aspnetcore/pull/48664#discussion_r1222680088

Reply:
I initially put it in `WebAssembly.Server` because it's consistent with how `AddServerComponents` is in `Components.Server`, and inverting the direction of the dependency requires moving `WebAssembly.Server` into the shared framework.

Does this align with your thinking, or was there another reason you think it belongs in `WebAssembly.Server`, @javiercn?